blog/2024-09-24-robotflow-reductstore/index.mdx

```diff
@@ -0,0 +1,397 @@
+
+![ReductStore for Efficient Data Management](./img/reductstore_pipeline.drawio.png)
+
+<small style={{ display: "block", marginTop: "-20px" }}>
+  ReductStore for Efficient Data Management. Image by author.
+</small>
+
+### ReductStore Setup
+
+Create a `docker-compose.yml` file with the following configuration:
+
+```yaml
+services:
+  reductstore:
+    image: reduct/store:latest
+    ports:
+      - "8383:8383"
+    volumes:
+      - data:/data
+    environment:
+      - RS_API_TOKEN=my-token
+      - RS_BUCKET_1_NAME=my-bucket-1
+      - RS_BUCKET_2_NAME=my-bucket-2
+
+volumes:
+  data:
+    driver: local
+```
+
+Then run the following command to start ReductStore:
+
+```bash
+docker compose up -d
+```
+
+This will start ReductStore on port 8383 with the specified API token and bucket names.
+
+![ReductStore Web Console](./img/reductstore_web_console.png)
+
+<small style={{ display: "block", marginTop: "-20px" }}>
+  ReductStore Web Console. Image source: [ReductStore](/)
+</small>
+
+### Storing Images with AI Labels at the Edge
+
+One of the key challenges in edge computing is managing large data sets generated by cameras or other imaging devices. ReductStore addresses this by
+
+1.  **Storing high-frequency image data**: Efficiently manages the storage of continuous streams of images captured at the edge. ReductStore is significantly faster than other time series databases for records with large binary payloads.
+2.  **AI-generated labels**: ReductStore supports the storage of metadata, such as object labels, directly with the image records, simplifying future analysis and retrieval of information.
+3.  **Efficient storage management**: Edge devices often have limited storage capacity. ReductStore provides the ability to set a FIFO (First In, First Out) policy based on volume to efficiently manage storage.
+
+```python
+import time
+from reduct import Client, Bucket
+
+async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
+    bucket: Bucket = await client.create_bucket("my-bucket-1", exist_ok=True)
+    ts = time.time()
+    await bucket.write(
+        "roboflow",
+        image_bytes,
+        ts,
+        labels=flat_predictions,
+        content_type="text/plain",
+    )
+```
+
+For more information on storing images with AI labels at the edge, refer to the [**ReductStore data ingestion guide**](https://www.reduct.store/docs/next/guides/data-ingestion).
+
+### Selective Data Replication to the Cloud
+
+While edge storage is critical for low-latency processing, cloud storage provides scalability and backup for long-term data retention. ReductStore provides selective data replication capabilities that allow users to
+
+1.  **Replicate only critical data**: Instead of sending all data to the cloud, developers can choose which records to replicate, significantly reducing cloud storage costs by focusing on critical data.
```

Comment:
maybe little explanation that a use can label critical data for replicating to  the cloud